consts.go

```diff
@@ -24,8 +24,13 @@ const (
 )
 
 const (
-	metadataHeader  = "WAL"
-	metadataVersion = "1.0"
+	metadataHeader  = "github.com/NebulousLabs/writeaheadlog\n"
+	metadataVersion = "v1.0.0 \n" // Extra room to allow for versions like v3.1.41
```

Comment:
why the newlines?

Reply:
Also, to give ourselves more headroom, why not make the version header "vXX.XX.XX"?

Reply:
Newlines make it more readable when put to disk, an mean that we don't have to insert the newlines manually when writing out the code.

Giving ourselves more room would be fine, but then we'd need to allocate a full 16 more bytes, since it's intentionally 16-byte aligned. Not super terrible, but there'd be 17 spaces and just 7 non-spaces.

Reply:
As it were, I find it highly unlikely that we ever drift into 9 character land, and if we do we can always strip out the `v`

Reply:
alternative: you could allow 48 (or maybe 64) bytes for header + version + status, and then use `\n` as a separator. Then you have some flexibility wrt how long the header is and how long the version is.

Reply:
tbh I liked it better when the metadata fields were constant size byte slices. You can still use `len` in the code instead of hardcoded values and `byte.Compare` is probably more lightweight than casting to a string and comparing.

Reply:
Switched to byte arrays, dropped the byte alignment (just used the number of bytes that made sense instead), dropped the constants we didn't need anymore, added enough spaces for vXX.XX.XX

Reply:
*kept the newlines though